packages/graphql-language-service-interface/src/GraphQLLanguageService.ts

```diff
@@ -76,7 +75,7 @@ export class GraphQLLanguageService {
     // Perform syntax diagnostics first, as this doesn't require
     // schema/fragment definitions, even the project configuration.
     let queryHasExtensions = false;
-    const projectConfig = this._graphQLConfig.getConfigForFile(uri);
+    const projectConfig = this._graphQLConfig.getConfigForFile(uri) as GraphQLProjectConfig;
```

Comment:
Prefer 
```suggestion
    const projectConfig: GraphQLProjectConfig = this._graphQLConfig.getConfigForFile(uri);
```

since the `as` syntax [looses type safety](https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions). It does seem that the type is being sourced from `graphql-config` so this should be safe?

https://github.com/graphql/graphiql/blob/6878d7c9c4bf9dc586b003b8effaa8d5f892cad4/packages/graphql-language-service-types/src/index.ts#L29

Reply:
thanks yeah, ill look into this some more. had to cast the config somehow, because graphql-config returns as Maybe<> essentially, which would give a bunch of errors where it's used. The type inference seemed to work either way but I will do this instead as a better practice

Reply:
unfortunately there isn't a way around this without breaking the refactor bubble, since the type we get back from `graphQLConfig.getConfigForFile()` will always give us `GraphQLConfig | undefined`, thus we need to check for whether it's undefined every time if we go that route.